Formula/kube-no-trouble.rb

```diff
@@ -0,0 +1,20 @@
+# Documentation: https://docs.brew.sh/Formula-Cookbook
+#                https://rubydoc.brew.sh/Formula
+# PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
+class KubeNoTrouble < Formula
+  desc "Easily check your cluster for use of deprecated APIs"
+  homepage "https://www.doit-intl.com"
+  url "https://github.com/doitintl/kube-no-trouble/archive/refs/tags/0.4.0.tar.gz"
+  sha256 "b9d63f41ae1b64b9dd161c1eaa3a0de4f35b5c0030793f491342d1c2058b042d"
+  license "MIT"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", *std_go_args, "-o", bin/"kubent", "cmd/kubent/main.go"
+  end
+
+  test do
+    system bin/"kubent", "-h"
```

Comment:
Can we have a functional test, and not one that checks help or version output? These rarely fail for broken installs. You could even have a test which errors out because a cluster isn't found.

Reply:
actually, the tool works only when there is a Kubernetes cluster; useless without it. 
not sure what type of test to create here

Reply:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.